staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -175,6 +179,10 @@ type WatchServer struct {
 	// used to encode the nested object in the watch stream
 	EmbeddedEncoder runtime.Encoder
 
+	// transformEvent an optional function that can transform the given
+	// object before sending it to a client.
+	transformEvent func(watch.Event, runtime.Encoder) watch.Event
```

Comment:
Rather than making it a function, we should instead create a new encoder itself.

The chance that something goes wrong is super long with just bookmark, but in general, having a transform function like that break the whole mechanism of "caching serializations".
(see https://github.com/kubernetes/kubernetes/pull/119801 + https://github.com/kubernetes/kubernetes/pull/119801)

Reply:
Maybe instead of creating a new encoder we should add a new transformer to the [watchEmbeddedEncoder](https://github.com/kubernetes/kubernetes/blob/0dd495e6dc253f94b0ad0bb92178fb5e8981116b/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go#L82)?

Reply:
For example https://github.com/kubernetes/kubernetes/pull/127411, see if you like it.